test/tests/preconditioners/hmg/tests

```diff
@@ -0,0 +1,29 @@
+[Tests]
+  design = 'HMGPreconditioner.md'
+  issues = '#16210'
+
+  [hmg]
+    type = 'Exodiff'
+    input = 'diffusion_hmg.i'
+    exodiff = 'diffusion_hmg_out.e'
+    min_parallel = 2
+    # HMG was introduced in 3.12.0
+    petsc_version = '>=3.12.0'
+    requirement = "The system shall support the use of HMG (high performance MG)"
+    # Check if PtAP algorithms are setup correctly
+    expect_out = "using allatonce MatPtAP\(\) implementation"
+  []
+
+  [hmg_3D]
+    type = 'Exodiff'
+    input = 'diffusion_hmg.i'
+    exodiff = 'diffusion_hmg_3d_out.e'
+    min_parallel = 2
+    # HMG was introduced in 3.12.0
+    petsc_version = '>=3.12.0'
+    cli_args = 'Mesh/dmg/dim=3 Mesh/dmg/nz=10 Outputs/file_base=diffusion_hmg_3d_out -log_view'
+    requirement = "The system shall support the use of HMG (high performance MG) for 3D problems"
+    # Check if strong_threshold is setup correctly
+    expect_out = "PETSc Preconditioner:    hmg strong_threshold: 0.7"
```

Comment:
This regex seems a little brittle. Can we use something like `\s+` instead of x number of spaces?

Reply:
Done